app/uk/gov/hmrc/excisemovementcontrolsystemapi/services/SubmissionMessageService.scala

```diff
@@ -43,11 +44,8 @@ class SubmissionMessageServiceImpl @Inject() (
     authorisedErn: String
   )(implicit hc: HeaderCarrier): Future[Either[EISErrorResponseDetails, EISSubmissionResponse]] = {
 
-    val correlationId = hc
-      .headers(Seq(HttpHeader.xCorrelationId))
-      .find(_._1 == HttpHeader.xCorrelationId)
-      .map(_._2)
-      .getOrElse(correlationIdService.generateCorrelationId())
+    val correlationId =
```

Comment:
Interested to understand why here we throw an exception if there's no correlationId, but in the NRS service we generate a new one?

Reply:
This was going to be default behaviour (I can make it regenerate instead) but for the tooling used by the scheduled jobs I was concerned that we'd just throw exceptions constantly. I can amend this to regenerate.

Reply:
In fact, considering the correlationId is only used to be passed to the connector, I'll introduce the logic to the connector instead